src/Effects.js

```diff
@@ -430,6 +430,113 @@ module.exports = class Effects {
       }
     };
 
+    this.kaleidoscope = {
+      init: function () {
+        this.shapes = p5.createGraphics(200, 200);
+        this.shapes.pixelDensity(1);
+        this.shapes.fill('white');
+        this.shapes.noStroke();
+        this.shapes.angleMode(p5.DEGREES);
+
+        this.hex = p5.createGraphics(200, 200);
+        this.hex.pixelDensity(1);
+        this.hex.angleMode(p5.DEGREES);
+
+        this.h = Math.sqrt(3) / 2 * 100;
+      },
+      blitHex: function () {
+        this.hex.push();
+        this.hex.clear();
+        this.hex.translate(100, 100);
+        this.hex.rotate(30);
+        for (let i = 0; i < 3; i++) {
+          this.hex.image(this.shapes, -50, 0);
+          this.hex.scale(-1, 1);
+          this.hex.rotate(60);
+          this.hex.image(this.shapes, -50, 0);
+          this.hex.rotate(60);
+          this.hex.scale(-1, 1);
+        }
+        this.hex.pop();
+      },
+      row: function (n) {
+        p5.push();
+        for (let i = 0; i < n; i++) {
+          p5.image(this.hex, 0, 0);
+          p5.translate(this.h * 2, 0);
+        }
+        p5.pop();
+      },
+      draw: function () {
+        if (!this.shapes) {
+          this.init();
+        }
+
+        p5.background('#333');
+
+        const ctx = this.shapes._renderer.drawingContext;
+        ctx.save();
+        ctx.beginPath();
+        ctx.moveTo(50, 0);
+        ctx.lineTo(100, this.h);
+        ctx.lineTo(0, this.h);
+        ctx.clip();
+        this.shapes.clear();
+        this.shapes.rotate(p5.frameCount);
```

Comment:
Possibly unneeded optimization: It looks like the only thing that varies on the `shapes` buffer with each frame is its rotation, so I'm wondering if it's possible to do all of the drawing to `shapes` just once, and get the rotation effect with a frame-dependent transform in the `blitHex` operation.

Reply:
I like this.  I think I'll hold off on this for now, in case we want to make the shapes change with the music.  But if we see perf issues this would be a great optimization!